server/sonar-main/src/main/java/org/sonar/application/command/CommandFactoryImpl.java

```diff
@@ -67,27 +68,71 @@ public CommandFactoryImpl(Props props, File tempDir, System2 system2) {
   }
 
   @Override
-  public EsCommand createEsCommand() {
-    EsFileSystem esFileSystem = new EsFileSystem(props);
-    if (!esFileSystem.getExecutable().exists()) {
+  public AbstractCommand<?> createEsCommand() {
+    if (IS_OS_WINDOWS) {
+      return createEsCommandForWindows();
+    }
+    return createEsCommandForNonWindows();
+  }
+
+  private ExternalCommand createEsCommandForNonWindows() {
```

Comment:
To keep things simple, I suggest to name this method `createEsCommandForUnix`

Reply:
Done.